function-memoization/src/main/java/com/tomtom/kotlin/FunctionMemoizationExtensions.kt

```diff
@@ -0,0 +1,141 @@
+/*
```

Comment:
Need to provide unit tests for all of these methods.

Reply:
All should be covered in `FunctionMemoizationExtensionsTest`. Is there anything I've missed?